src/main/frontend/components/views.cljs

```diff
@@ -1255,13 +1250,93 @@
+       (shui/select-item {:value "desc"} "Descending"))))
+    (shui/button
+     {:variant "ghost"
+      :class "text-muted-foreground !px-1"
+      :size :sm
+      :on-click (fn []
+                  (let [f (:column-set-sorting! table)
+                        new-sorting (f sorting {:id id} nil)
+                        f (get-in table [:data-fns :set-sorting!])]
+                    (set-sorting! new-sorting)
+                    (f new-sorting)
+                    (when (empty? new-sorting)
+                      (shui/popup-hide!))))}
+     (ui/icon "x"))]])
+
+(rum/defc view-sorting-config
+  [table sorting columns]
+  (let [[sorting set-sorting!] (rum/use-state sorting)]
+    [:div.ls-view-order-setting.flex.flex-col.gap-2.py-2.text-sm
+     (let [items (for [{:keys [id asc?]} sorting]
+                   (when-let [name (some (fn [column] (when (= id (:id column))
+                                                        (:name column))) columns)]
+                     {:id (str id)
+                      :value id
+                      :content (view-sorting-item table sorting id name asc? set-sorting!)}))]
+       (dnd/items items
+                  {:on-drag-end (fn [ordered-columns]
+                                  (let [f (get-in table [:data-fns :set-sorting!])
+                                        new-sorting (mapv (fn [column] (some #(when (= column (:id %)) %) sorting)) ordered-columns)]
+                                    (set-sorting! new-sorting)
+                                    (f new-sorting)))}))
+     (shui/dropdown-menu-item
+      {:class "text-muted-foreground"
+       :on-click (fn []
+                   (let [f (get-in table [:data-fns :set-sorting!])]
+                     (set-sorting! nil)
+                     (f nil)
+                     (shui/popup-hide!)))}
+      (ui/icon "trash" {:size 15})
+      [:span.ml-1 "Delete sort"])]))
+
+(rum/defc view-sorting
+  [table columns sorting]
+  (shui/button
+   {:variant "ghost"
+    :class "text-muted-foreground !px-1"
+    :size :sm
+    :on-click (fn [e]
+                (shui/popup-show! (.-target e)
+                                  (fn [] (view-sorting-config table sorting columns))
+                                  {:align :end}))}
+   (ui/icon "arrows-up-down")))
+
 (rum/defc ^:large-vars/cleanup-todo view-inner < rum/static
   [view-entity {:keys [data set-data! columns add-new-object! views-title title-key render-empty-title?] :as option
                 :or {render-empty-title? false}}
    *scroller-ref]
   (let [[input set-input!] (rum/use-state "")
-        sorting (:logseq.property.table/sorting view-entity)
-        [sorting set-sorting!] (rum/use-state (or sorting [{:id :block/updated-at, :asc? false}]))
+        sorting* (:logseq.property.table/sorting view-entity)
+        sorting (if (= sorting* :logseq.property/empty-placeholder) nil sorting*)
+        [sorting set-sorting!] (rum/use-state sorting)
```

Comment:
With multi-sorting enabled, the first time a user interacted with any table they were adding a 2nd sort so I removed the default :block/updated-at as it felt buggy. I think users expect the first time they use a sort on a table for it take affect as the only sort. I [took a video](https://www.loom.com/share/6c9cdcb917ad4684af8f1ba0669c5c9d) of the behavior that seemed confusing. Feel free to change if you think of a better solution